src/main/java/scrolls/elder/logic/commands/UndoCommand.java

```diff
@@ -0,0 +1,26 @@
+package scrolls.elder.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import scrolls.elder.logic.commands.exceptions.CommandException;
+import scrolls.elder.model.Model;
+
+/**
+ * Reverses the effect of a previous operation on the Datastore.
+ */
+public class UndoCommand extends Command {
+    public static final String COMMAND_WORD = "undo";
+    public static final String MESSAGE_SUCCESS = "Previous operation has been undone!";
+    public static final String MESSAGE_UNDO_ERROR = "No previous operation to be undone";
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        if (!model.getDatastoreVersionStorage().canUndo()) {
+            throw new CommandException(MESSAGE_UNDO_ERROR);
+        }
+
+        model.undoChanges();
+        return new CommandResult(MESSAGE_SUCCESS);
```

Comment:
Same as RedoCommand, should we update the displayed list after undoing the operation?

Reply:
That's a good point, maybe it would be be clearer for the user as well, I'll add that in.